### ISSUE TITLE
what is Termux (failsafe)?

### ISSUE THREAD

**Issue by dyseo** [termux/termux-app#1101]
I have been using this termux for a long time, but recently I found something strange!
suddenly a new application appears called, *Termux (failsafe)* on my android, I don't know this is a bug or something I just want to delete this but when I delete it the two termux are deleted and after I reinstall termux, and *Termux (failsafe)* remains appear..

I don't know how to deal with this, very grateful to anyone who can help

> Reading the discussion in the mentioned PR you'll see that the only way to remove the `failsafe` icon is by hiding it from the launcher.
> 
> If that's an issue for you, maybe you should create a new issue.

i have solved my issues using termux from F-droid, thanks for answering

**Comment by brunelli** [comment]
See https://github.com/termux/termux-app/issues/277#issuecomment-286309177 and #1028.

**Comment by dyseo** [comment]
> See [#277 (comment)](https://github.com/termux/termux-app/issues/277#issuecomment-286309177) and #1028.

i have read that comment, and i try too but still not working for me

**Comment by brunelli** [comment]
Reading the discussion in the mentioned PR you'll see that the only way to remove the `failsafe` icon is by hiding it from the launcher.

If that's an issue for you, maybe you should create a new issue.

**Comment by xeffyr** [comment]
We have made sessions auto-closeable, see https://github.com/termux/termux-app/issues/988 for more information.

Auto-closeable sessions may put application into "denial-of-service" condition if user will mess with files like .bashrc - in such case access to files on internal storage will be completely lost. Thus we had to create a separate icon.

It's very unlikely that we will remove it. Easy access to failsafe shell is mandatory but unfortunately there no other *reliable* variants that will work on all Android versions.
